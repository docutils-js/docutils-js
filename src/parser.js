import Parser from './parsers/restructuredtext';
import newDocument from './newDocument';
import baseSettings from './baseSettings';

function parse(docSource, settings) {
    const parser = new Parser({});
    if(typeof settings === 'undefined') {
	settings = { ... baseSettings };
    }
    const document = newDocument({sourcePath: ''}, settings);
    parser.parse(docSource, document);
    return document;    
}
