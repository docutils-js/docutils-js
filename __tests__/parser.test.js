import parser from '../src/index';

test('parse misc doc', () => {
    expect(parser.parse(`* hello\n*test\n\nButter\n======\n\n    I am a test.\n`)).toMatchSnapshot();
});
