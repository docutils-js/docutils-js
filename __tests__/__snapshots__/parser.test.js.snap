// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse misc doc 1`] = `
document {
  "attributes": Object {
    "backrefs": Array [],
    "classes": Array [],
    "dupnames": Array [],
    "ids": Array [],
    "names": Array [],
    "source": "",
  },
  "autofootnoteRefs": Array [],
  "autofootnoteStart": 1,
  "autofootnotes": Array [],
  "basicAttributes": Array [
    "ids",
    "classes",
    "names",
    "dupnames",
  ],
  "childTextSeparator": "

",
  "children": Array [
    bullet_list {
      "attributes": Object {
        "backrefs": Array [],
        "bullet": "*",
        "classes": Array [],
        "dupnames": Array [],
        "ids": Array [],
        "names": Array [],
      },
      "basicAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
      ],
      "childTextSeparator": "

",
      "children": Array [
        list_item {
          "attributes": Object {
            "backrefs": Array [],
            "classes": Array [],
            "dupnames": Array [],
            "ids": Array [],
            "names": Array [],
          },
          "basicAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
          ],
          "childTextSeparator": "

",
          "children": Array [
            paragraph {
              "attributes": Object {
                "backrefs": Array [],
                "classes": Array [],
                "dupnames": Array [],
                "ids": Array [],
                "names": Array [],
              },
              "basicAttributes": Array [
                "ids",
                "classes",
                "names",
                "dupnames",
              ],
              "childTextSeparator": "

",
              "children": Array [
                Text {
                  "children": Array [],
                  "data": "hello",
                  "document": undefined,
                  "line": undefined,
                  "parent": [Circular],
                  "rawsource": "hello",
                  "source": undefined,
                  "tagname": "Text",
                },
              ],
              "document": undefined,
              "knownAttributes": Array [
                "ids",
                "classes",
                "names",
                "dupnames",
                "backrefs",
                "source",
                "rawsource",
              ],
              "line": 1,
              "listAttributes": Array [
                "ids",
                "classes",
                "names",
                "dupnames",
                "backrefs",
              ],
              "localAttributes": Array [
                "backrefs",
              ],
              "nodeName": Symbol(Element),
              "parent": [Circular],
              "source": undefined,
              "tagname": "paragraph",
            },
          ],
          "document": undefined,
          "knownAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
            "source",
            "rawsource",
          ],
          "line": undefined,
          "listAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
          ],
          "localAttributes": Array [
            "backrefs",
          ],
          "nodeName": Symbol(Element),
          "parent": [Circular],
          "source": undefined,
          "tagname": "list_item",
        },
      ],
      "document": undefined,
      "knownAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
        "backrefs",
        "source",
        "rawsource",
      ],
      "line": 1,
      "listAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
        "backrefs",
      ],
      "localAttributes": Array [
        "backrefs",
      ],
      "nodeName": Symbol(Element),
      "parent": [Circular],
      "source": undefined,
      "tagname": "bullet_list",
    },
    system_message {
      "addBackref": [Function],
      "attributes": Object {
        "backrefs": Array [],
        "classes": Array [],
        "dupnames": Array [],
        "ids": Array [],
        "line": 2,
        "names": Array [],
      },
      "basicAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
      ],
      "childTextSeparator": "

",
      "children": Array [
        paragraph {
          "attributes": Object {
            "backrefs": Array [],
            "classes": Array [],
            "dupnames": Array [],
            "ids": Array [],
            "names": Array [],
          },
          "basicAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
          ],
          "childTextSeparator": "

",
          "children": Array [
            Text {
              "children": Array [],
              "data": "Bullet list ends without a blank line; unexpected unindent.",
              "document": undefined,
              "line": undefined,
              "parent": [Circular],
              "rawsource": "",
              "source": undefined,
              "tagname": "Text",
            },
          ],
          "document": undefined,
          "knownAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
            "source",
            "rawsource",
          ],
          "line": undefined,
          "listAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
          ],
          "localAttributes": Array [
            "backrefs",
          ],
          "nodeName": Symbol(Element),
          "parent": [Circular],
          "source": undefined,
          "tagname": "paragraph",
        },
      ],
      "document": undefined,
      "knownAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
        "backrefs",
        "source",
        "rawsource",
      ],
      "line": 1,
      "listAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
        "backrefs",
      ],
      "localAttributes": Array [
        "backrefs",
      ],
      "nodeName": Symbol(Element),
      "parent": [Circular],
      "source": undefined,
      "tagname": "system_message",
    },
    paragraph {
      "attributes": Object {
        "backrefs": Array [],
        "classes": Array [],
        "dupnames": Array [],
        "ids": Array [],
        "names": Array [],
      },
      "basicAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
      ],
      "childTextSeparator": "

",
      "children": Array [
        problematic {
          "attributes": Object {
            "backrefs": Array [],
            "classes": Array [],
            "dupnames": Array [],
            "ids": Array [
              "id2",
            ],
            "names": Array [],
            "refid": "id1",
          },
          "basicAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
          ],
          "childTextSeparator": "

",
          "children": Array [
            Text {
              "children": Array [],
              "data": "*",
              "document": undefined,
              "line": undefined,
              "parent": [Circular],
              "rawsource": "",
              "source": undefined,
              "tagname": "Text",
            },
          ],
          "document": undefined,
          "knownAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
            "source",
            "rawsource",
          ],
          "line": undefined,
          "listAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
          ],
          "localAttributes": Array [
            "backrefs",
          ],
          "nodeName": Symbol(Element),
          "parent": [Circular],
          "source": undefined,
          "tagname": "problematic",
        },
        Text {
          "children": Array [],
          "data": "test",
          "document": undefined,
          "line": undefined,
          "parent": [Circular],
          "rawsource": "test",
          "source": undefined,
          "tagname": "Text",
        },
      ],
      "document": undefined,
      "knownAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
        "backrefs",
        "source",
        "rawsource",
      ],
      "line": 2,
      "listAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
        "backrefs",
      ],
      "localAttributes": Array [
        "backrefs",
      ],
      "nodeName": Symbol(Element),
      "parent": [Circular],
      "source": undefined,
      "tagname": "paragraph",
    },
    system_message {
      "addBackref": [Function],
      "attributes": Object {
        "backrefs": Array [
          "id2",
        ],
        "classes": Array [],
        "dupnames": Array [],
        "ids": Array [
          "id1",
        ],
        "line": 2,
        "names": Array [],
      },
      "basicAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
      ],
      "childTextSeparator": "

",
      "children": Array [
        paragraph {
          "attributes": Object {
            "backrefs": Array [],
            "classes": Array [],
            "dupnames": Array [],
            "ids": Array [],
            "names": Array [],
          },
          "basicAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
          ],
          "childTextSeparator": "

",
          "children": Array [
            Text {
              "children": Array [],
              "data": "Inline Function start-string without end-string.",
              "document": undefined,
              "line": undefined,
              "parent": [Circular],
              "rawsource": "",
              "source": undefined,
              "tagname": "Text",
            },
          ],
          "document": undefined,
          "knownAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
            "source",
            "rawsource",
          ],
          "line": undefined,
          "listAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
          ],
          "localAttributes": Array [
            "backrefs",
          ],
          "nodeName": Symbol(Element),
          "parent": [Circular],
          "source": undefined,
          "tagname": "paragraph",
        },
      ],
      "document": undefined,
      "knownAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
        "backrefs",
        "source",
        "rawsource",
      ],
      "line": 3,
      "listAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
        "backrefs",
      ],
      "localAttributes": Array [
        "backrefs",
      ],
      "nodeName": Symbol(Element),
      "parent": [Circular],
      "source": undefined,
      "tagname": "system_message",
    },
    section {
      "attributes": Object {
        "backrefs": Array [],
        "classes": Array [],
        "dupnames": Array [],
        "ids": Array [
          "butter",
        ],
        "names": Array [
          "butter",
        ],
      },
      "basicAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
      ],
      "childTextSeparator": "

",
      "children": Array [
        title {
          "attributes": Object {
            "backrefs": Array [],
            "classes": Array [],
            "dupnames": Array [],
            "ids": Array [],
            "names": Array [],
          },
          "basicAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
          ],
          "childTextSeparator": "

",
          "children": Array [
            Text {
              "children": Array [],
              "data": "Butter",
              "document": undefined,
              "line": undefined,
              "parent": [Circular],
              "rawsource": "Butter",
              "source": undefined,
              "tagname": "Text",
            },
          ],
          "document": undefined,
          "knownAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
            "source",
            "rawsource",
          ],
          "line": undefined,
          "listAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
          ],
          "localAttributes": Array [
            "backrefs",
          ],
          "nodeName": Symbol(Element),
          "parent": [Circular],
          "source": undefined,
          "tagname": "title",
        },
        block_quote {
          "attributes": Object {
            "backrefs": Array [],
            "classes": Array [],
            "dupnames": Array [],
            "ids": Array [],
            "names": Array [],
          },
          "basicAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
          ],
          "childTextSeparator": "

",
          "children": Array [
            paragraph {
              "attributes": Object {
                "backrefs": Array [],
                "classes": Array [],
                "dupnames": Array [],
                "ids": Array [],
                "names": Array [],
              },
              "basicAttributes": Array [
                "ids",
                "classes",
                "names",
                "dupnames",
              ],
              "childTextSeparator": "

",
              "children": Array [
                Text {
                  "children": Array [],
                  "data": "I am a test.",
                  "document": undefined,
                  "line": undefined,
                  "parent": [Circular],
                  "rawsource": "I am a test.",
                  "source": undefined,
                  "tagname": "Text",
                },
              ],
              "document": undefined,
              "knownAttributes": Array [
                "ids",
                "classes",
                "names",
                "dupnames",
                "backrefs",
                "source",
                "rawsource",
              ],
              "line": 1,
              "listAttributes": Array [
                "ids",
                "classes",
                "names",
                "dupnames",
                "backrefs",
              ],
              "localAttributes": Array [
                "backrefs",
              ],
              "nodeName": Symbol(Element),
              "parent": [Circular],
              "source": undefined,
              "tagname": "paragraph",
            },
          ],
          "document": undefined,
          "knownAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
            "source",
            "rawsource",
          ],
          "line": undefined,
          "listAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
          ],
          "localAttributes": Array [
            "backrefs",
          ],
          "nodeName": Symbol(Element),
          "parent": [Circular],
          "source": undefined,
          "tagname": "block_quote",
        },
      ],
      "document": undefined,
      "knownAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
        "backrefs",
        "source",
        "rawsource",
      ],
      "line": 5,
      "listAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
        "backrefs",
      ],
      "localAttributes": Array [
        "backrefs",
      ],
      "nodeName": Symbol(Element),
      "parent": [Circular],
      "source": undefined,
      "tagname": "section",
    },
  ],
  "citationRefs": Object {},
  "citations": Array [],
  "currentLine": undefined,
  "currentSource": undefined,
  "decoration": undefined,
  "document": [Circular],
  "footnoteRefs": Object {},
  "footnotes": Array [],
  "idStart": 3,
  "ids": Object {
    "butter": section {
      "attributes": Object {
        "backrefs": Array [],
        "classes": Array [],
        "dupnames": Array [],
        "ids": Array [
          "butter",
        ],
        "names": Array [
          "butter",
        ],
      },
      "basicAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
      ],
      "childTextSeparator": "

",
      "children": Array [
        title {
          "attributes": Object {
            "backrefs": Array [],
            "classes": Array [],
            "dupnames": Array [],
            "ids": Array [],
            "names": Array [],
          },
          "basicAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
          ],
          "childTextSeparator": "

",
          "children": Array [
            Text {
              "children": Array [],
              "data": "Butter",
              "document": undefined,
              "line": undefined,
              "parent": [Circular],
              "rawsource": "Butter",
              "source": undefined,
              "tagname": "Text",
            },
          ],
          "document": undefined,
          "knownAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
            "source",
            "rawsource",
          ],
          "line": undefined,
          "listAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
          ],
          "localAttributes": Array [
            "backrefs",
          ],
          "nodeName": Symbol(Element),
          "parent": [Circular],
          "source": undefined,
          "tagname": "title",
        },
        block_quote {
          "attributes": Object {
            "backrefs": Array [],
            "classes": Array [],
            "dupnames": Array [],
            "ids": Array [],
            "names": Array [],
          },
          "basicAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
          ],
          "childTextSeparator": "

",
          "children": Array [
            paragraph {
              "attributes": Object {
                "backrefs": Array [],
                "classes": Array [],
                "dupnames": Array [],
                "ids": Array [],
                "names": Array [],
              },
              "basicAttributes": Array [
                "ids",
                "classes",
                "names",
                "dupnames",
              ],
              "childTextSeparator": "

",
              "children": Array [
                Text {
                  "children": Array [],
                  "data": "I am a test.",
                  "document": undefined,
                  "line": undefined,
                  "parent": [Circular],
                  "rawsource": "I am a test.",
                  "source": undefined,
                  "tagname": "Text",
                },
              ],
              "document": undefined,
              "knownAttributes": Array [
                "ids",
                "classes",
                "names",
                "dupnames",
                "backrefs",
                "source",
                "rawsource",
              ],
              "line": 1,
              "listAttributes": Array [
                "ids",
                "classes",
                "names",
                "dupnames",
                "backrefs",
              ],
              "localAttributes": Array [
                "backrefs",
              ],
              "nodeName": Symbol(Element),
              "parent": [Circular],
              "source": undefined,
              "tagname": "paragraph",
            },
          ],
          "document": undefined,
          "knownAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
            "source",
            "rawsource",
          ],
          "line": undefined,
          "listAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
          ],
          "localAttributes": Array [
            "backrefs",
          ],
          "nodeName": Symbol(Element),
          "parent": [Circular],
          "source": undefined,
          "tagname": "block_quote",
        },
      ],
      "document": undefined,
      "knownAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
        "backrefs",
        "source",
        "rawsource",
      ],
      "line": 5,
      "listAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
        "backrefs",
      ],
      "localAttributes": Array [
        "backrefs",
      ],
      "nodeName": Symbol(Element),
      "parent": [Circular],
      "source": undefined,
      "tagname": "section",
    },
    "id1": system_message {
      "addBackref": [Function],
      "attributes": Object {
        "backrefs": Array [
          "id2",
        ],
        "classes": Array [],
        "dupnames": Array [],
        "ids": Array [
          "id1",
        ],
        "line": 2,
        "names": Array [],
      },
      "basicAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
      ],
      "childTextSeparator": "

",
      "children": Array [
        paragraph {
          "attributes": Object {
            "backrefs": Array [],
            "classes": Array [],
            "dupnames": Array [],
            "ids": Array [],
            "names": Array [],
          },
          "basicAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
          ],
          "childTextSeparator": "

",
          "children": Array [
            Text {
              "children": Array [],
              "data": "Inline Function start-string without end-string.",
              "document": undefined,
              "line": undefined,
              "parent": [Circular],
              "rawsource": "",
              "source": undefined,
              "tagname": "Text",
            },
          ],
          "document": undefined,
          "knownAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
            "source",
            "rawsource",
          ],
          "line": undefined,
          "listAttributes": Array [
            "ids",
            "classes",
            "names",
            "dupnames",
            "backrefs",
          ],
          "localAttributes": Array [
            "backrefs",
          ],
          "nodeName": Symbol(Element),
          "parent": [Circular],
          "source": undefined,
          "tagname": "paragraph",
        },
      ],
      "document": undefined,
      "knownAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
        "backrefs",
        "source",
        "rawsource",
      ],
      "line": 3,
      "listAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
        "backrefs",
      ],
      "localAttributes": Array [
        "backrefs",
      ],
      "nodeName": Symbol(Element),
      "parent": [Circular],
      "source": undefined,
      "tagname": "system_message",
    },
    "id2": problematic {
      "attributes": Object {
        "backrefs": Array [],
        "classes": Array [],
        "dupnames": Array [],
        "ids": Array [
          "id2",
        ],
        "names": Array [],
        "refid": "id1",
      },
      "basicAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
      ],
      "childTextSeparator": "

",
      "children": Array [
        Text {
          "children": Array [],
          "data": "*",
          "document": undefined,
          "line": undefined,
          "parent": [Circular],
          "rawsource": "",
          "source": undefined,
          "tagname": "Text",
        },
      ],
      "document": undefined,
      "knownAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
        "backrefs",
        "source",
        "rawsource",
      ],
      "line": undefined,
      "listAttributes": Array [
        "ids",
        "classes",
        "names",
        "dupnames",
        "backrefs",
      ],
      "localAttributes": Array [
        "backrefs",
      ],
      "nodeName": Symbol(Element),
      "parent": paragraph {
        "attributes": Object {
          "backrefs": Array [],
          "classes": Array [],
          "dupnames": Array [],
          "ids": Array [],
          "names": Array [],
        },
        "basicAttributes": Array [
          "ids",
          "classes",
          "names",
          "dupnames",
        ],
        "childTextSeparator": "

",
        "children": Array [
          [Circular],
          Text {
            "children": Array [],
            "data": "test",
            "document": undefined,
            "line": undefined,
            "parent": [Circular],
            "rawsource": "test",
            "source": undefined,
            "tagname": "Text",
          },
        ],
        "document": undefined,
        "knownAttributes": Array [
          "ids",
          "classes",
          "names",
          "dupnames",
          "backrefs",
          "source",
          "rawsource",
        ],
        "line": 2,
        "listAttributes": Array [
          "ids",
          "classes",
          "names",
          "dupnames",
          "backrefs",
        ],
        "localAttributes": Array [
          "backrefs",
        ],
        "nodeName": Symbol(Element),
        "parent": [Circular],
        "source": undefined,
        "tagname": "paragraph",
      },
      "source": undefined,
      "tagname": "problematic",
    },
  },
  "indirectTargets": Array [],
  "knownAttributes": Array [
    "ids",
    "classes",
    "names",
    "dupnames",
    "backrefs",
    "source",
    "rawsource",
  ],
  "line": undefined,
  "listAttributes": Array [
    "ids",
    "classes",
    "names",
    "dupnames",
    "backrefs",
  ],
  "localAttributes": Array [
    "backrefs",
  ],
  "nameIds": Object {
    "butter": "butter",
  },
  "nameTypes": Object {
    "butter": undefined,
  },
  "nodeName": Symbol(Element),
  "parent": undefined,
  "parseMesssages": Array [],
  "refIds": Object {},
  "refNames": Object {},
  "reporter": Reporter {
    "DEBUG_LEVEL": 0,
    "ERROR_LEVEL": 3,
    "INFO_LEVEL": 1,
    "SEVERE_LEVEL": 4,
    "WARNING_LEVEL": 2,
    "debugFlag": false,
    "encoding": "utf-8",
    "errorHandler": "backslashreplace",
    "getSourceAndLine": [Function],
    "haltLevel": 4,
    "maxLevel": 2,
    "observers": Array [],
    "reportLevel": 2,
    "source": "",
    "stream": null,
  },
  "settings": Object {
    "autoIdPrefix": "id",
    "debug": false,
    "dumpInternals": null,
    "dumpPsuedoXml": null,
    "dumpSettings": null,
    "dumpTransforms": null,
    "errorEncoding": "utf-8",
    "errorEncodingErrorHandler": "backslashreplace",
    "exitStatusLevel": 5,
    "exposeInternalAttribute": null,
    "footnoteBacklinks": true,
    "haltLevel": 4,
    "idPrefix": "",
    "initialHeaderLevel": 1,
    "inputEncodingErrorHandler": "strict",
    "languageCode": "en",
    "mathOutput": "HTML math.css",
    "outputEncoding": "utf-8",
    "outputEncodingErrorHandler": "strict",
    "recordDependencies": null,
    "reportLevel": 2,
    "sectionNumbering": true,
    "strictVisitor": null,
    "tocBacklinks": "entry",
    "traceback": null,
    "warningStream": null,
  },
  "source": undefined,
  "substitutionDefs": Object {},
  "substitutionNames": Object {},
  "symbolFootnoteRefs": Array [],
  "symbolFootnoteStart": 0,
  "symbolFootnotes": Array [],
  "tagname": "document",
  "transformMessages": Array [],
  "transformer": Transformer {
    "applied": Array [],
    "components": Object {},
    "document": [Circular],
    "serialno": 0,
    "sorted": 0,
    "transforms": Array [],
    "unknownReferenceResolvers": Array [],
  },
}
`;
